### PR TITLE
Update longformer.md

### PR DESCRIPTION
**Pull request by JihadHammoud02**
Refactored Longformer docs
Added examples for pipeline, Automodel and cli
Added quantization
Did not add a Attention visualizer, from what I researched it doesn't support it, if it is not the case I am happy to add it !
Added a note concerning versions < 4.37.0.dev

### CODE REVIEW

File: docs/source/en/model_doc/longformer.md [hunk 1]
```diff
@@ -9,93 +8,144 @@ Unless required by applicable law or agreed to in writing, software distributed
...
+    </div>
 </div>
 
-## Overview
-
-The Longformer model was presented in [Longformer: The Long-Document Transformer](https://arxiv.org/pdf/2004.05150.pdf) by Iz Beltagy, Matthew E. Peters, Arman Cohan.
+# Longformer
 
-The abstract from the paper is the following:
+[Longformer](https://huggingface.co/papers/2004.05150) is a transformer model designed for long-document processing. It introduces a novel attention mechanism that scales linearly with sequence length by combining local windowed attention with task-specific global attention, enabling efficient processing of documents with thousands of tokens.
```
Comment (stevhliu) [inline docs/source/en/model_doc/longformer.md:24]:
```suggestion
[Longformer](https://huggingface.co/papers/2004.05150) is a transformer model designed for processing long documents. The self-attention operation usually scales quadratically with sequence length, preventing transformers from processing longer sequences. The Longformer attention mechanism overcomes this by scaling linearly with sequence length. It combines local windowed attention with task-specific global attention, enabling efficient processing of documents with thousands of tokens.

You can find all the original Longformer checkpoints under the [Ai2](https://huggingface.co/allenai?search_models=longformer) organization.
```

File: docs/source/en/model_doc/longformer.md [hunk 2]
```diff
@@ -9,93 +8,144 @@ Unless required by applicable law or agreed to in writing, software distributed
...
-windowed attention with a task motivated global attention. Following prior work on long-sequence transformers, we
-evaluate Longformer on character-level language modeling and achieve state-of-the-art results on text8 and enwik8. In
-contrast to most prior work, we also pretrain Longformer and finetune it on a variety of downstream tasks. Our
-pretrained Longformer consistently outperforms RoBERTa on long document tasks and sets new state-of-the-art results on
-WikiHop and TriviaQA.*
+> [!TIP]
+> Click on the Longformer models in the right sidebar for more examples of how to apply Longformer to different language tasks.
 
-This model was contributed by [beltagy](https://huggingface.co/beltagy). The Authors' code can be found [here](https://github.com/allenai/longformer).
+The example below demonstrates how to generate text with [`Pipeline`], [`AutoModel`] and from the command line.
```
Comment (stevhliu) [inline docs/source/en/model_doc/longformer.md:29]:
```suggestion
The example below demonstrates how to fill the `<mask>` token with [`Pipeline`], [`AutoModel`] and from the command line.
```

File: docs/source/en/model_doc/longformer.md [hunk 3]
```diff
@@ -9,93 +8,144 @@ Unless required by applicable law or agreed to in writing, software distributed
...
+echo -e "San Francisco 49ers cornerback Shawntae Spencer will miss the rest of the <mask> with a torn ligament in his left knee." | transformers-cli run --task fill-mask --model allenai/longformer-base-4096 --device 0
+```
 
-- 0: the token attends "locally",
-- 1: the token attends "globally".
+</hfoption>
+</hfoptions
 
-For more information please also refer to [`~LongformerModel.forward`] method.
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
```
Comment (stevhliu) [inline docs/source/en/model_doc/longformer.md:86]: I don't think we need a Quantization example here since the model isn't that big

File: docs/source/en/model_doc/longformer.md [hunk 4]
```diff
@@ -9,93 +8,144 @@ Unless required by applicable law or agreed to in writing, software distributed
...
 
-## Resources
+## Notes
 
-- [Text classification task guide](../tasks/sequence_classification)
-- [Token classification task guide](../tasks/token_classification)
-- [Question answering task guide](../tasks/question_answering)
-- [Masked language modeling task guide](../tasks/masked_language_modeling)
-- [Multiple choice task guide](../tasks/multiple_choice)
+- If you're using Transformers < 4.37.0.dev, set `trust_remote_code=True` in [~AutoModel.from_pretrained]. Otherwise, make sure you update Transformers to the latest stable version.
```
Comment (stevhliu) [inline docs/source/en/model_doc/longformer.md:121]: Not necessary to include this note. Instead, add the below

- Longformer is based on [RoBERTa](./roberta) and doesn't have `token_type_ids`. You don't need to indicate which token belongs to which segment. You only need to separate the segments with the separation token `</s>` or `tokenizer.sep_token`.
- You can set which tokens can attend *locally* and which tokens attend *globally* with the `global_attention_mask` at inference (see this [example](https://huggingface.co/docs/transformers/en/model_doc/longformer#transformers.LongformerModel.forward.example) for more details). A value of `0` means a token attends locally and a value of `1` means a token attends globally.
- [`LongformerForMaskedLM`] is trained like [`RobertaForMaskedLM`] and should be used as shown below.
   ```py
   input_ids = tokenizer.encode("This is a sentence from [MASK] training data", return_tensors="pt")
   mlm_labels = tokenizer.encode("This is a sentence from the training data", return_tensors="pt")
   loss = model(input_ids, labels=input_ids, masked_lm_labels=mlm_labels)[0]
   ```

**Review by stevhliu — COMMENTED** [review]
Nice, thanks for adding!

**Review by stevhliu — APPROVED** [review]
Thanks again! 🤗
